File: algorandsamples/java-test/src/main/java/com/algorand/javatest/indexer/SearchTransactionsPaging.java
//SearchTransactionsPaging.java
package com.algorand.javatest.indexer;

import com.algorand.algosdk.v2.client.common.IndexerClient;
import com.algorand.algosdk.v2.client.common.Client;

import org.json.JSONArray;
import org.json.JSONObject;

	

public class SearchTransactionsPaging {
    public Client indexerInstance = null;
    // utility function to connect to a node
    private Client connectToNetwork(){
        final String INDEXER_API_ADDR = "localhost";
        final int INDEXER_API_PORT = 8980;       
        IndexerClient indexerClient = new IndexerClient(INDEXER_API_ADDR, INDEXER_API_PORT); 
        return indexerClient;
    }

    public static void main(String args[]) throws Exception {
        SearchTransactionsPaging ex = new SearchTransactionsPaging();
        IndexerClient indexerClientInstance = (IndexerClient) ex.connectToNetwork();
        String nexttoken = "";
        Integer numtx = 1;
        String responseall = new String();
        
        //loop until there are no more tranactions in the response
        //for the limit (max is 1000  per request)
        //"min_amount": 100000000000000,
        while (numtx > 0) {
            Long min_amount = Long.valueOf(100000000000000L);
            Long limit = Long.valueOf(10);
            String next_page = nexttoken;
            String response = indexerClientInstance.searchForTransactions().next(next_page)
                    .currencyGreaterThan(min_amount).limit(limit).execute().toString();
            JSONObject jsonObj = new JSONObject(response.toString());

            JSONArray jsonArray = (JSONArray) jsonObj.get("transactions");
            numtx = jsonArray.length();
            if (numtx > 0) {

                nexttoken = jsonObj.get("next-token").toString();

                responseall = responseall.toString() + response.toString();

            }
        }

        // { replace ] with , }
        //    responseall = "{" + responseall + "}";
        JSONObject jsonObjAll = new JSONObject(responseall.toString());

        //     JSONArray jsonArrayAll = (JSONArray) jsonObjAll.get("transactions");
        // String response = indexerClientInstance.searchForTransactions().next(next_page).currencyGreaterThan(min_amount).limit(limit).execute().toString();
        //  JSONObject jsonObj = new JSONObject(responseall.toString());
        System.out.println("Transaction Info: " + jsonObjAll.toString(2)); // pretty print json
    }
 }

File: algorandsamples/v2/myjsdemo/indexer/searchTransactionsPaging.js
// SearchTransactionsPaging.js
// requires algosdk@1.6.1 or higher 
// verify installed version
// npm list algosdk

const algosdk = require('algosdk');
const indexer_token = "";
const indexer_server = "http://localhost";
const indexer_port = 8980;
// const indexer_server = "https://testnet-algorand.api.purestake.io/idx2/";
// const indexer_port = "";
// const indexer_token = {
//     'X-API-key': 'B3SU4KcVKi94Jap2VXkK83xx38bsv95K5UZm2lab',
// }

// Instantiate the indexer client wrapper
let indexerClient = new algosdk.Indexer(indexer_token, indexer_server, indexer_port);

let nexttoken = "";
let numtx = 1;
// loop until there are no more transactions in the response
// for the limit(max limit is 1000  per request)    
(async () => {
    let min_amount = 500000000000;
    let limit = 2;
    while (numtx > 0) {
        // execute code as long as condition is true
        let next_page = nexttoken;
        let response = await indexerClient.searchForTransactions()
            .limit(limit)
            .currencyGreaterThan(min_amount)
            .nextToken(next_page).do();
        let transactions = response['transactions'];
        numtx = transactions.length;
        if (numtx > 0)
        {
            nexttoken = response['next-token']; 
            console.log("Transaction Information: " + JSON.stringify(response, undefined, 2));           
        }
    }
})().catch(e => {
    console.log(e);
    console.trace();
});